File: src/com/l2kt/gameserver/network/clientpackets/CannotMoveAnymore.java
package com.l2kt.gameserver.network.clientpackets;

import com.l2kt.gameserver.model.actor.Creature;
import com.l2kt.gameserver.model.actor.ai.CtrlEvent;
import com.l2kt.gameserver.model.location.SpawnLocation;

public final class CannotMoveAnymore extends L2GameClientPacket
{
	private int _x;
	private int _y;
	private int _z;
	private int _heading;
	
	@Override
	protected void readImpl()
	{
		_x = readD();
		_y = readD();
		_z = readD();
		_heading = readD();
	}
	
	@Override
	protected void runImpl()
	{
		final Creature player = getClient().getActiveChar();
		if (player == null)
			return;
		
		if (player.hasAI())
			player.getAI().notifyEvent(CtrlEvent.EVT_ARRIVED_BLOCKED, new SpawnLocation(_x, _y, _z, _heading));
	}
}

File: src/com/l2kt/gameserver/network/clientpackets/CannotMoveAnymore.kt
package com.l2kt.gameserver.network.clientpackets

import com.l2kt.gameserver.model.actor.ai.CtrlEvent
import com.l2kt.gameserver.model.location.SpawnLocation

class CannotMoveAnymore : L2GameClientPacket() {
    private var _x: Int = 0
    private var _y: Int = 0
    private var _z: Int = 0
    private var _heading: Int = 0

    override fun readImpl() {
        _x = readD()
        _y = readD()
        _z = readD()
        _heading = readD()
    }

    override fun runImpl() {
        val player = client.activeChar ?: return

        if (player.hasAI())
            player.ai.notifyEvent(CtrlEvent.EVT_ARRIVED_BLOCKED, SpawnLocation(_x, _y, _z, _heading))
    }
}